### PR TITLE
feat(dev): Add object matcher pytest fixture

### PR DESCRIPTION
**Pull request by lobsterkatie**
Adds the ability to test against partial objects. Also adds tests for previously-added matcher fixtures.

### CODE REVIEW

File: tests/test_conftest.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+@pytest.mark.parametrize(
+    "test_string, expected_result",
+    [
+        # type matches
+        ("dogs are great!", True),  # full containment - beginning
+        ("go, dogs, go!", True),  # full containment - middle
+        ("I like dogs", True),  # full containment - end
+        ("dogs", True),  # equality
+        ("", False),  # reverse containment
+        ("dog", False),  # reverse conainment
```
Comment (dashed) [inline tests/test_conftest.py:13]:
```suggestion
        ("dog", False),  # reverse containment
```

File: tests/test_conftest.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+
+@pytest.mark.parametrize(
+    "test_dict, expected_result",
+    [
+        # type matches
+        ({"dogs": "yes", "cats": "maybe", "spiders": "nope"}, True),  # full containment
+        ({"dogs": "yes", "cats": "maybe"}, True),  # equality
+        ({}, False),  # reverse containment
+        ({"dogs": "yes"}, False),  # reverse conainment
```
Comment (dashed) [inline tests/test_conftest.py:38]:
```suggestion
        ({"dogs": "yes"}, False),  # reverse containment
```

File: tests/test_conftest.py (hunk 1 above)
Comment (lobsterkatie) [reply tests/test_conftest.py:13]: 🤦🏻‍♀️

Thanks for catching.

**Review by wedamija — APPROVED** [review]

File: tests/conftest.py [hunk 3]
```diff
@@ -391,6 +394,67 @@ def __eq__(self, test_dict):
...
 
-            return all(test_dict.get(key) == self.subdict[key] for key in self.subdict)
+            if len(self.subdict) > len(test_dict):
+                return False
+
+            # Have to test self == other (rather than vice-versa) in case
+            # any of the values in self.subdict is another matcher with a custom
+            # __eq__ method (in LHS == RHS, LHS's __eq__ is tried before RHS's).
+            # In other words, this order is important so that examples like
+            # {"dogs", "are great"} == DictionaryContaining({"dogs": StringContaining("great")})
```
[the remark this answers is not in the document]
Comment (lobsterkatie) [reply tests/conftest.py:404]: Shoot. Yes. Thanks.

**Pull request merged**